### PR TITLE
add pylint configuration and package

### PR DESCRIPTION
**Pull request by jurraca**
pylintrc generated with `--generate-rcfile` which outputs a ton of default config. I removed the empty settings and kept the stated defaults for the rest. Our disabled rules form CI are on line 209. Would it be better to remove the default config settings?

### CODE REVIEW

File: pylintrc [hunk 1]
```diff
@@ -0,0 +1,417 @@
...
+        too-many-locals,
+        unspecified-encoding,
+        too-many-return-statements,
+        too-many-branches,
+        too-many-nested-blocks,
+        too-many-instance-attributes,
+        too-many-statements,
+        too-few-public-methods,
+        fixme,
+        too-many-positional,
```
Comment (fjahr) [inline pylintrc:233]: Getting this error here:

```
pylintrc:1:0: W0012: Unknown option value for '--disable', expected a valid pylint message and got 'too-many-positional' (unknown-option-value)
```

**Review by fjahr — COMMENTED** [review]
Seems fine for me to keep the defaults in but also fine to remove them, I have no strong opinion either way.

You should also update the CI pylint command to use the rc file in this PR so that the disablements are managed in one place.

File: pylintrc (hunk 1 above)
Comment (jurraca) [reply pylintrc:233]: what pylint version you running? Looks like nixpkgs is on 3.1.1 (fairly far behind) which has this option, but 4.0.0 has `too-many-positional-arguments` [arg](https://pylint.pycqa.org/en/latest/user_guide/messages/refactor/too-many-positional-arguments.html). 4.0.0 would be the version we should follow in this case.

Comment (jurraca) [reply pylintrc:233]: nixpkgs `24.11` release is around the corner (currently in beta) and has 3.3.1. I'll test it.

Comment (fjahr) [reply pylintrc:233]: seems like I tested this outside of the nix shell and there I have pylint 3.3.1, so yeah, using that should fix it.

Comment (jurraca) [reply pylintrc]: updated and tested locally with 3.3.1 :+1:

**Comment by fjahr** [comment]
tACK 8dd314d6d8ecbc5dcdde03b08a4f29cf2cbb2a6f

Awesome, thanks!

**Pull request merged**